packages/web/src/uiUtils.ts

```diff
@@ -141,42 +141,64 @@ export function copy(text: string) {
 
 export type Setting<T> = readonly [T, (value: T) => void, string];
 
-export function useSetting<S>(name: string | undefined, defaultValue: S, title?: string): [S, React.Dispatch<React.SetStateAction<S>>, Setting<S>] {
-  if (name)
-    defaultValue = settings.getObject(name, defaultValue);
-  const [value, setValue] = React.useState<S>(defaultValue);
-  const setValueWrapper = React.useCallback((value: React.SetStateAction<S>) => {
-    if (name)
-      settings.setObject(name, value);
-    setValue(value);
-  }, [name, setValue]);
+export function useSetting<S>(name: string, defaultValue: S, title?: string, dontPersist?: boolean): [S, (v: S) => void, Setting<S>] {
+  const value = React.useSyncExternalStore(
+      function subscribe(callback) {
```

Comment:
React recommends passing the same `subscribe` function instance every time. Perhaps `useCallback` for this?